src/account/operations/output_claiming.rs

```diff
@@ -225,93 +230,82 @@ impl AccountHandle {
-                            new_native_tokens.add_native_tokens(native_tokens.clone())?;
-                        }
+                    new_amount += output_data.output.amount();
+                    if let Some(native_tokens) = output_data.output.native_tokens() {
+                        new_native_tokens.add_native_tokens(native_tokens.clone())?;
                     }
                 }
             }
 
-            // Check if the new amount is enough for the storage deposit, otherwise increase it to this
             let option_native_token = if new_native_tokens.is_empty() {
                 None
             } else {
                 Some(new_native_tokens.clone().finish()?)
             };
+
+            // Check if the new amount is enough for the storage deposit, otherwise increase it to this
             let required_storage_deposit = minimum_storage_deposit(
                 &byte_cost_config,
                 &first_account_address.address.inner,
                 &option_native_token,
             )?;
-            let mut additional_inputs = Vec::new();
 
+            let mut additional_inputs = Vec::new();
             if new_amount < required_storage_deposit {
                 // add more inputs
                 for output_data in &possible_additional_inputs {
                     // Recalculate every time, because new intputs can also add more native tokens, which would increase
                     // the storage deposit cost
-                    let required_storage_deposit = minimum_storage_deposit(
+                    let mut required_storage_deposit = minimum_storage_deposit(
                         &byte_cost_config,
                         &first_account_address.address.inner,
                         &option_native_token,
                     )?;
                     if new_amount < required_storage_deposit {
-                        new_amount += output_data.output.amount();
-                        if let Some(native_tokens) = output_data.output.native_tokens() {
-                            new_native_tokens.add_native_tokens(native_tokens.clone())?;
+                        if !additional_inputs_used.contains(&output_data.output_id) {
+                            new_amount += output_data.output.amount();
+                            if let Some(native_tokens) = output_data.output.native_tokens() {
+                                new_native_tokens.add_native_tokens(native_tokens.clone())?;
+                                // Re-calculate the required storage deposit amount since we have added a new native
+                                // token
+                                required_storage_deposit = minimum_storage_deposit(
+                                    &byte_cost_config,
+                                    &first_account_address.address.inner,
+                                    &option_native_token,
+                                )?;
+                            }
```

Comment:
Hm we do that in 288 already each time before we compare, so I guess this is not needed